marketplace/plugins/awsredshift/lib/index.ts

```diff
@@ -0,0 +1,81 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-marketplace/common';
+import { SourceOptions, QueryOptions } from './types';
+import {
+  RedshiftDataClient,
+  ExecuteStatementCommand,
+  GetStatementResultCommand,
+  DescribeStatementCommand,
+} from '@aws-sdk/client-redshift-data';
+
+export default class Awsredshift implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
+    try {
+      const input = {
+        Sql: queryOptions.sql_query,
+        Database: sourceOptions.database,
+        WithEvent: false,
+        WorkgroupName: sourceOptions.workgroup_name,
+      };
+      const isSelectQuery = queryOptions.sql_query.trim().toLowerCase().startsWith('select');
+      const client = await this.getConnection(sourceOptions);
+      const command = new ExecuteStatementCommand(input);
+      const executeResult = await client.send(command);
+
+      //Polling for the above query
+      if (isSelectQuery) {
+        const describeInput = { Id: executeResult.Id };
+        let status = 'SUBMITTED';
+        while (status !== 'FINISHED') {
```

Comment:
Do add a timeout for polling to break out of the while loop.
We can add a 30s timeout.